libs/vkd3d/command.c

```diff
@@ -1681,6 +1681,284 @@ static void d3d12_command_list_invalidate_current_pipeline(struct d3d12_command_
+    {
+        stages = VK_PIPELINE_STAGE_EARLY_FRAGMENT_TESTS_BIT | VK_PIPELINE_STAGE_LATE_FRAGMENT_TESTS_BIT;
+        access = VK_ACCESS_DEPTH_STENCIL_ATTACHMENT_READ_BIT | VK_ACCESS_DEPTH_STENCIL_ATTACHMENT_WRITE_BIT;
+        subpass_desc.pDepthStencilAttachment = &attachment_ref;
+    }
+    else
+    {
+        stages = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;
+        access = VK_ACCESS_COLOR_ATTACHMENT_READ_BIT | VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT;
+        subpass_desc.colorAttachmentCount = 1;
+        subpass_desc.pColorAttachments = &attachment_ref;
+    }
+
+    dependencies[0].srcSubpass = VK_SUBPASS_EXTERNAL;
+    dependencies[0].dstSubpass = 0;
+    dependencies[0].srcStageMask = stages;
+    dependencies[0].dstStageMask = stages;
+    dependencies[0].srcAccessMask = 0;
+    dependencies[0].dstAccessMask = access;
+    dependencies[0].dependencyFlags = 0;
+
+    dependencies[1].srcSubpass = 0;
+    dependencies[1].dstSubpass = VK_SUBPASS_EXTERNAL;
+    dependencies[1].srcStageMask = stages;
+    dependencies[1].dstStageMask = stages;
+    dependencies[1].srcAccessMask = access;
+    dependencies[1].dstAccessMask = 0;
+    dependencies[1].dependencyFlags = 0;
+
+    pass_info.sType = VK_STRUCTURE_TYPE_RENDER_PASS_CREATE_INFO;
+    pass_info.pNext = NULL;
+    pass_info.flags = 0;
+    pass_info.attachmentCount = 1;
+    pass_info.pAttachments = &attachment_desc;
+    pass_info.subpassCount = 1;
+    pass_info.pSubpasses = &subpass_desc;
+    pass_info.dependencyCount = ARRAY_SIZE(dependencies);
+    pass_info.pDependencies = dependencies;
+
+    if ((vr = VK_CALL(vkCreateRenderPass(list->device->vk_device, &pass_info, NULL, &vk_render_pass))) < 0)
+    {
+        WARN("Failed to create Vulkan render pass, vr %d.\n", vr);
+        return;
+    }
+
+    if (!d3d12_command_allocator_add_render_pass(list->allocator, vk_render_pass))
+    {
+        WARN("Failed to add render pass.\n");
+        VK_CALL(vkDestroyRenderPass(list->device->vk_device, vk_render_pass, NULL));
+        return;
+    }
+
+    if (!d3d12_command_allocator_add_view(list->allocator, view))
+        WARN("Failed to add view.\n");
+
+    extent.width = d3d12_resource_desc_get_width(&resource->desc, view->info.texture.miplevel_idx);
+    extent.height = d3d12_resource_desc_get_height(&resource->desc, view->info.texture.miplevel_idx);
+    extent.depth = view->info.texture.layer_count;
+
+    if (!d3d12_command_list_create_framebuffer(list, vk_render_pass, 1, &view->u.vk_image_view, extent, &vk_framebuffer))
+    {
+        ERR("Failed to create framebuffer.\n");
+        return;
+    }
+
+    begin_info.sType = VK_STRUCTURE_TYPE_RENDER_PASS_BEGIN_INFO;
+    begin_info.pNext = NULL;
+    begin_info.renderPass = vk_render_pass;
+    begin_info.framebuffer = vk_framebuffer;
+    begin_info.renderArea.offset.x = 0;
+    begin_info.renderArea.offset.y = 0;
+    begin_info.renderArea.extent.width = extent.width;
```

Comment:
We can still use clear. LOAD_OP_CLEAR only affects renderArea.

Reply:
Doable, but somewhat annoying since we can only really so it if we have *one* clear rect that *doesn't* cover the entire image, and we can't transition the image anyway. Is that really worth it over just doing `vkCmdClearAttachments` in that case?

Reply:
Right, for multiple rects, this gets rather annoying. Okay to do it like this I guess. Partial clear rect should be rare.